dominion/src/dominion/arena.rs

```diff
@@ -115,88 +119,103 @@ impl Arena {
         }
     }
 
-    // This function is context dependent
-    pub fn select_card(
-        &mut self,
-        player_id: usize,
-        card: CardKind,
-        location: Location,
-    ) -> Result<()> {
+    pub fn play_action(&mut self, player_id: usize, card: CardKind) -> Result<()> {
         self.check_active(player_id)?;
 
+        let card_index = self.players[player_id]
+            .hand
+            .iter()
```

Comment:
This method is doing two things: 
1) find the first index of a card with the given kind in a players' hand.
2) play the gard at the given index.

I can imagine 1 being useful when writing bots but at other times we will already know the index and just need 2.

Reply:
I suggest to put 1 as a convenience method of the player or leave it out of the core library and just take a card hand index in this method.

Reply:
I think you're right about there being situations where we will already know the index and just need to play the card at the given index but I'll put off changing this until that situation comes up.

Reply:
In all of the test cases you know the card index because you just push it into the hand. 

Lets say you're writing a bot and you want it to play two cards if they're both in the hand. You can't do this with the current API without searching through your hands unnecessarily.